.github/workflows/db.yml

```diff
@@ -0,0 +1,256 @@
+  COMPOSER_VERSION: 2.7.1
+
+jobs:
+  migrations-mysql:
+    name: MySQL / MariaDB
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      # See: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
+      matrix:
+        php_version:
+          - 8.3
+        db_version:
+          - mariadb:latest
+          - mariadb:11
+          - mariadb:10
+
+          - mysql:latest
+          - mysql:8.3
+          - mysql:8.0
+
+    container:
+      image: php:${{ matrix.php_version }}-cli
+
+    services:
+      db:
+        image: ${{ matrix.db_version }}
+        env:
+          MARIADB_ROOT_PASSWORD: password
+          MARIADB_PASSWORD: password
+          MARIADB_USER: pixelfed
+          MARIADB_DATABASE: pixelfed
+
+          MYSQL_ROOT_PASSWORD: "password"
+          MYSQL_USER: "pixelfed"
+          MYSQL_PASSWORD: "password"
+          MYSQL_DATABASE: "pixelfed"
+        ports:
+          - 33306:3306
+
+      redis:
+        image: redis:7.2
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    # See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-and-the-default-behavior
+    concurrency:
+      group: db-migrations-${{ github.ref }}-${{ matrix.php_version }}-${{matrix.db_version}}
+      cancel-in-progress: true
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Download mlocati/docker-php-extension-installer
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: mlocati/docker-php-extension-installer
+          tag: ${{env.DOCKER_PHP_EXTENSION_INSTALLER_VERSION}}
+          fileName: install-php-extensions
+
+      - name: Download composer
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: composer/composer
+          tag: ${{env.COMPOSER_VERSION}}
+          fileName: composer.phar
+
+      - name: Download jippi/dottie
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: jippi/dottie
+          latest: true
+          fileName: "*_amd64.deb"
+
+      - name: Install dottie
+        run: |
+          dpkg -i *.deb
+          rm -f *.deb
+          dottie -v
+
+      - name: Install PHP extensions
+        run: |
+          chmod +x install-php-extensions
+          ./install-php-extensions ${{env.EXTRA_PHP_EXTENSIONS}}
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v4
+        with:
+          path: vendor/
+          key: composer-${{matrix.php_version}}-${{ hashFiles('composer.lock') }}
+
+      - name: Install composer dependencies
+        run: php composer.phar install -n --prefer-dist
+
+      - name: Setup Environment
+        run: |
+          cp .env.testing .env
+
+          dottie set DB_CONNECTION=mysql
```

Comment:
Shouldn't this be `${{matrix.db}}` but stripping the tag from it?

Reply:
there is a couple of differences between the MySQL and PostgreSQL jobs - and while it can be merged, I think the complexity of that isn't worth taking on vs a more verbose, but simple-to-follow step. CI should be as dumb as possible to keep it easy to debug and flexible.